jujugui/static/gui/src/app/components/lightbox/lightbox.js

```diff
@@ -1,35 +1,155 @@
-/* Copyright (C) 2015 Canonical Ltd. */
-
+/* Copyright (C) 2017 Canonical Ltd. */
 'use strict';
 
 const PropTypes = require('prop-types');
 const React = require('react');
+const classNames = require('classnames');
 
 const SvgIcon = require('../svg-icon/svg-icon');
 
 /**
   Renders a modal like overlay with a darkened background.
   A caption is displayed in white below the content if present.
 */
-class Lightbox extends React.PureComponent {
+class Lightbox extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      activeSlide: 0,
+      lastSlide: null
```

Comment:
Is this here so you don't have to check the length of the children multiple times?

Reply:
Yeah, it also means that if there is one child - and no navigation or slides are needed - I don't have to keep checking for `this.props.children.map` or any other way to check it's an array.